lib/commit-lint.ts

```diff
@@ -0,0 +1,152 @@
+import { commentFunction } from "./ci-helper";
+import { IPRCommit } from "./github-glue";
+
+/*
+ * Simple class to drive lint tests on commit messages.
+ */
+export class LintCommit {
+    private readonly addComment: commentFunction;
+
+    public constructor(addComment: commentFunction) {
+        this.addComment = addComment;
+    }
+
+    /*
+     * Linter method to run checks on the commit message.
+     * @param {IPRCommit} the patch to be checked
+     *
+     * The lintings are in methods called from here.  If
+     * a lint check is too severe to continue, it will throw
+     * an error.
+     */
+    public async lint(patch: IPRCommit):
+        Promise<boolean> {
+        let result: boolean = true;
+        const lines =  patch.message.split("\n");
+        const msgs: string[] = [];
+
+        try {
+           // add a line for each test
+
+            // tslint:disable:curly
+           if (!await this.lint1(lines, msgs)) result = false;
+           if (!await this.lint2(lines, msgs)) result = false;
+           if (!await this.lint3(lines, msgs)) result = false;
+           if (!await this.lint4(lines, msgs)) result = false;
```

Comment:
As I mentioned, I would like to make it the duty of the _caller_ to keep track of `result`.

Also, can we please use more descriptive names for the methods?

Finally, I think that this could be made easier to read by using a combination of `Promise.all()` and the `.bind()` method, i.e. something like:

```javascript
await Promise.all([
    commitMessageLength, lowerCaseAfterPrefix, signedOffBy, moreThanAHyperlink,
].map((method) => async {
    await (method.bind(this, lines, msgs)).();
});
```

Reply:
Let me know if the changes resolve this.

Reply:
Yes!